From c7e62da6e569fec03baf43c9a6953849eddb2160 Mon Sep 17 00:00:00 2001
From: Kevin Darbyshire-Bryant <ldir@darbyshire-bryant.me.uk>
Date: Thu, 16 Mar 2023 19:23:26 +0530
Subject: [PATCH 108/500] OpenWrt: 211-darwin-uuid-typedef-clash.patch

file2alias: build on macos

Signed-off-by: Kevin Darbyshire-Bryant <ldir@darbyshire-bryant.me.uk>

Change-Id: I4cc3d210d42584bca83b1ee2d593540b8efc57ae
Signed-off-by: Ram Chandra Jangir <quic_rjangir@quicinc.com>
(cherry picked from commit 01259d02be6971b5a2a04ce3a91b08dea433f132)
---
 scripts/mod/file2alias.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/scripts/mod/file2alias.c b/scripts/mod/file2alias.c
index 6583b36dbe69..271e1159f449 100644
--- a/scripts/mod/file2alias.c
+++ b/scripts/mod/file2alias.c
@@ -35,6 +35,9 @@ typedef uint32_t	__u32;
 typedef uint16_t	__u16;
 typedef unsigned char	__u8;
 
+#ifdef __APPLE__
+#define uuid_t compat_uuid_t
+#endif
 /* UUID types for backward compatibility, don't use in new code */
 typedef struct {
 	__u8 b[16];
-- 
2.34.1

